### PR TITLE
[Merged by Bors] - Port/data.pnat.find

### PR DESCRIPTION
**Pull request by arienmalec**
Port of data.pnat.find

Also changed `Nat.find_x` to `Nat.findX` to follow naming conventions

### CODE REVIEW

**Comment by arienmalec** [comment]
Note that the simpNF doesn't like `one_le_find`; moving it up and solving it explicitly doesn't making the linter happy though.

File: Mathlib/Data/PNat/Find.lean [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    PNat.find h = PNat.find h₂ + 1 := by
+  refine' (find_eq_iff _).2 ⟨PNat.find_spec h₂, fun n => PNat.recOn n _ _⟩
+  · simp [h1]
+  intro m _ hm
+  simp only [add_lt_add_iff_right, lt_find_iff] at hm
+  exact hm _ le_rfl
+#align pnat.find_comp_succ PNat.find_comp_succ
+
+end PNat
+#lint
```
Comment (ChrisHughes24) [inline Mathlib/Data/PNat/Find.lean:133]:
```suggestion

```

**Comment by ChrisHughes24** [comment]
How about sticking to `findX` and changing the `Nat` version at the same time?

**Comment by arienmalec** [comment]
> How about sticking to `findX` and changing the `Nat` version at the same time?

done.

File: Mathlib/Init/Data/Nat/Lemmas.lean [hunk 2]
```diff
@@ -256,7 +256,7 @@ private def wf_lbp : WellFounded (lbp p) := by
   | zero => exact absurd pn (a _ kn)
   | succ m IH => exact IH _ (by rw [Nat.add_right_comm]; exact kn)
```
Comment (ChrisHughes24) [inline Mathlib/Init/Data/Nat/Lemmas.lean:258]:
```suggestion
Used in the definition of `Nat.find`. Returns the smallest natural satisfying `p`
```

File: Mathlib/Init/Data/Nat/Lemmas.lean [hunk 3]
```diff
@@ -255,8 +255,8 @@ private def wf_lbp : WellFounded (lbp p) := by
   induction m with refine fun k kn ↦ ⟨_, fun | _, ⟨rfl, a⟩ => ?_⟩
   | zero => exact absurd pn (a _ kn)
   | succ m IH => exact IH _ (by rw [Nat.add_right_comm]; exact kn)
-
-protected def find_x : {n // p n ∧ ∀ m, m < n → ¬p m} :=
+Used in the definition of `Nat.find`. Returns the smallest natural satisfying `p`
```
Comment (ChrisHughes24) [inline Mathlib/Init/Data/Nat/Lemmas.lean:258]:
```suggestion
/-- Used in the definition of `Nat.find`. Returns the smallest natural satisfying `p`-/
```

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**